src/state/actividades.js

```diff
@@ -1,16 +1,33 @@
-import { selector, selectorFamily } from 'recoil';
+import { apiById, apiIndex } from './api';
+
 import { DateTime } from 'luxon';
 import { dateFormatter } from '../utils/dateUtils';
-import { apiById, apiIndex } from './api';
+import queryString from 'query-string';
+import { selectorFamily } from 'recoil';
 
-export const todasLasActividades = selector({
+export const todasLasActividades = selectorFamily({
   key: 'todasLasActividades',
-  get: async ({ get }) => {
-    const { data } = get(apiIndex('actividades'));
+  get: (fecha) => async ({ get }) => {
+    const { data } = get(apiIndex(buildPath(fecha)));
     return data;
   },
 });
 
+const buildPath = (fecha) => {
+  const path = 'actividades';
+  if (fecha !== null) {
+    const query = queryString.stringify(
+      {
+        desde: fecha.desde,
+        hasta: fecha.hasta,
+      },
+      { skipNull: true }
+    );
+    return `${path}/?${query}`;
+  }
+  return path;
+};
```

Comment:
Lo que podés hacer acá para evitarte el chequeo de `null` es ponerle un valor por default. Quedaría así:

```suggestion
const buildPath = ({ desde, hasta } = {}) => {
  const query = queryString.stringify(
    { desde, hasta },
    { skipNull: true }
  );
  return `actividades?${query}`;
};
```

Y luego donde lo uses y no quieras poner filtro, no haría falta pasar `null`.